android/src/main/java/com/cuhacking/atlas/MainActivity.kt

```diff
@@ -7,22 +7,24 @@ import android.widget.SearchView
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.content.ContextCompat
-import androidx.lifecycle.Observer
+import androidx.lifecycle.lifecycleScope
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.cuhacking.atlas.databinding.ActivityMainBinding
-import com.cuhacking.atlas.search.SearchResult
 import com.cuhacking.atlas.search.SearchResultsAdapter
-import com.cuhacking.atlas.search.SearchViewModel
+import com.cuhacking.atlas.common.SearchViewModel
+import com.cuhacking.atlas.db.database
 import com.cuhacking.mapbox.GeoJsonSource
 import com.mapbox.mapboxsdk.maps.Style
 import com.mapbox.mapboxsdk.style.layers.FillLayer
 import com.mapbox.mapboxsdk.style.layers.LineLayer
 import com.mapbox.mapboxsdk.style.layers.PropertyFactory
 import io.github.dellisd.spatialk.geojson.FeatureCollection.Companion.toFeatureCollection
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
 class MainActivity : AppCompatActivity() {
     private lateinit var binding: ActivityMainBinding
-    private val searchViewModel = SearchViewModel()
+    private val searchViewModel = SearchViewModel(database)
```

Comment:
```suggestion
    private val searchViewModel = SearchViewModel(database, CoroutineDispatchers)
```
Related to another change below.